exchanges/stream/websocket.go

```diff
@@ -1241,3 +1263,37 @@ func signalReceived(ch chan struct{}) bool {
 		return false
 	}
 }
+
+// GetConnection returns a connection by connection signature (defined in wrapper setup) for request and response
+// handling in a multi connection context.
+func (w *Websocket) GetConnection(connSignature any) (Connection, error) {
+	if w == nil {
+		return nil, fmt.Errorf("%w: %T", common.ErrNilPointer, w)
+	}
+
+	if connSignature == nil {
+		return nil, errConnectionSignatureNotSet
+	}
+
+	w.m.Lock()
+	defer w.m.Unlock()
+
+	if !w.useMultiConnectionManagement {
+		return nil, fmt.Errorf("%s: multi connection management not enabled %w please use exported Conn and AuthConn fields", w.exchangeName, errCannotObtainOutboundConnection)
+	}
+
+	if !w.IsConnected() {
+		return nil, ErrNotConnected
+	}
+
+	for _, wrapper := range w.connectionManager {
+		if wrapper.Setup.WrapperDefinedConnectionSignature == connSignature {
```

Comment:
Can we have a type implement a "IsAppropriate" function instead?
It'd give us more flexibility, avoid reflection, etc.
So all your examples for asset type would use AssetFilter(asset.Spot), for example, and that'd be a type which implements `IsAppropriate`

Reply:
Can you send me a diff?

Reply:
Sure. Will do.

Reply:
Okay. I'm wrong.

Direction was this: https://github.com/gbjk/gocryptotrader/commit/33bb31e4d

But we're just allowing a function to compare X to Y, in which case we could use exchange defined struct types ( or even stream package defined, like UrlAssetFilter )

We don't need to make it a method call because I can't see a case where it wouldn't just be filtering on equality, yet.

So, my only change request is to sanitise this name to `MessageFilter` or similar. `WrapperDefined` prefix is wrong, not least because Wrapper isn't actually a thing, other than a file suffix _convention_. But also because we don't case what defined or provided the filter. We just care that we have one.

Reply:
[fixed](https://github.com/thrasher-corp/gocryptotrader/pull/1598/commits/cb965218c36e27c0547c98ccd2a2c9cb7e6ecb16)